docs/developer/index.rst

```diff
@@ -103,9 +103,25 @@ Run doctests locally
 If `doctestplus <https://pypi.org/project/pytest-doctestplus/>` is installed
 (and it probably is because it's part of
 `sphinx-astropy <https://pypi.org/project/sphinx-astropy/>`,
-which is required to build the documentation locally), you can
-run the examples in the documentation or in the doctrings of individual
-functions, for example:
+which is required to build the documentation locally),
+examples in the documentation are run automatically.
+This serves two purposes:
+
+  - It ensure that the examples we give execute and give the expected output.
+    It is confusing to a user to see an example in the documentation that does not
+    work because it contains bugs to calls functions that have been removed from sherpa.
+  - This also acts as an additional tests.
+
+However, many examples were written because doctestplus was used with Sherpa and do not execute
+as written (e.g. they need implicit set-up that's not spelled out),
+so a number of rst files are explicitly excluded from that run in `pyests.ini`.
+When an rst file is fixed to work top-to-bottom, it should be removed from the
+blacklist in `pytest.ini`.
+
+doctestplus can also run the examples in the doctrings of individual
+functions, but again many files are still blacklisted for this functionality in
+`pytest.ini`.
+During development, you can run doctestplus on individual files like so::
```

Comment:
This needs some work. Let me suggest updates after the coffee has kicked in and then left my system...